src/coreclr/vm/profilingenumerators.cpp

```diff
@@ -88,6 +89,44 @@ BOOL ProfilerFunctionEnum::Init(BOOL fWithReJITIDs)
     return TRUE;
 }
 
+// ---------------------------------------------------------------------------------------
+//  ProfilerObjectEnum/ICorProfilerObjectEnum implementation
+// ---------------------------------------------------------------------------------------
+
+BOOL ProfilerObjectEnum::Init()
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        CAN_TAKE_LOCK;
+    }
+    CONTRACTL_END;
+
+    GCX_COOP();
```

Comment:
What portion of the implementation requires us to switch to coop mode? This might make the API challenging to use without a deadlock. The primary use I am anticipating from profiler vendors is that they would enumerate both the GC and non-GC heaps during a GC. This means they would invoke this API inside one of the callbacks that GC is starting or finishing and at that point the runtime would be suspended. I'm guessing the transition to COOP mode would block until the EE resumes, the EE won't resume until the profiler GC callback returns, and the GC callback won't return until this enumeration completes.

Reply:
It's not needed indeed, removed